lib/levensthein-distance.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Gives you a array with filled with 0...amount - 1.
+ * @param {number} amount
+ * @returns {number[]}
+ */
+function range(amount) {
+  const range = Array(amount);
+  for (let i = 0; i < amount; i++) {
+    range[i] = i;
+  }
+
+  return range;
+}
+
+/**
+ * Check if the amount of edits between firstString and secondString is <= maxEdits.
+ * It uses the Levenshtein distance algorithm with the two matrix rows variant.
+ * @param {string} firstString First string to be checked against the other string
+ * @param {string} secondString Second string to be checked against the other string
+ * @param {number} maxEdit The maximum amount of edits that these 2 string should have.
+ * @returns {boolean} indicate if the 2 strings's edits are less or equal to maxEdits
+ */
+function LevenshteinDistanceWithMax(firstString, secondString, maxEdit) {
+  const lenOne = firstString.length;
+  const lenTwo = secondString.length;
+
+  const lenDiff = Math.abs(lenOne - lenTwo);
+  // Are the difference between 2 lengths greater than
+  // maxEdit, we know to bail out early on.
+  if (lenDiff > maxEdit) {
+    return false;
+  }
+
+  let prevRowDistance = range(lenOne + 1);
+  let currentRowDistance = Array(lenOne + 1);
+  for (let i = 1; i <= lenTwo; i++) {
+    // Calculate the current row distances from the previous row.
+    currentRowDistance[0] = i;
```

Comment:
A small improvement: since the distance function is a non-decreasing function, we can bail out once the minimum distance in the current row have passed `maxEdit`.
```js
let minDistance = currentRowDistance[0] = i;
for (...) {
  ...
  currentRowDistance[j] = ...
  if (currentRowDistance[j] < minDistance) {
    minDistance = currentRowDistance[j];
  }
}
if (minDistance > maxEdit) {
  return false;
}
```

Reply:
Seems good to me. And still passes the test 👍🏽.

I remember I had such thing in the first commit but due to switching implementation it got lost, 😄.

BTW, you might want to revisit the eslint rules, because `let minDistance = currentRowDistance[0] = i;` will conflict with the `multi-assign` rule.